src/blog/jwt-identification-and-hot-streams/index.md

```diff
@@ -0,0 +1,172 @@
+# AnyCable PRO:<br>JWT identification and "hot streams"
+
+September 27, 2021
+{date}
+
+When we started collecting applications for the [AnyCable PRO][pro] early access program back in July, we asked respondents to share the way they use _cables_ and which features they would like us to provide out-of-the-box. The analysis of their responses led to the development of two new PRO features, [JWT identification][jwt-id-docs] and [Signed streams support][signed-streams-docs], which might seem unrelated to each other from the first look but turned out to bring a huge performance boost when used together. Continue reading to learn more!
+{intro}
+
+<div class="divider"></div>
+
+## Dealing with authentication
+
+Among the requested features, one of the leaders was "a token-based authentication". Most Action Cable applications relies on cookies as an authentication mechanism (directly or via sessions). The main benefit of the cookie-based authentication is the simplicity: it just works. However, there are some drawbacks:
+
+- Authenticating non-web clients (e.g., mobile apps) is becoming cumbersome.
+- Rails API-only apps usually do not use tokens for web clients authentication as well (they use tokens).
+- WebSockets have no CORS support and, thus, **vulnerable to cross-site request forgery** (or [cross-site WebSocket hijacking][cross-site-ws-hijack]).
+
+Using tokens could help us to solve these problems. What's the cost? Well, we have to build everything ourselves, both server and client side (and good luck with finding a library or gem which makes it easier not harder).
```

Comment:
Using tokens could certainly help us solve these problems–but what's the catch? Well, we'd have to build everything ourselves, both the server and the client side (and good luck actually finding a library or gem which makes this process easier instead of harder).